### PR TITLE
Fix database locks

### PR DESCRIPTION
**Pull request by miaulalala**
See #4812

Added the timestamp to the hasLocks method as to avoid only relying on tokens but not checking if they are still valid.

### CODE REVIEW

File: lib/Db/MailboxMapper.php [hunk 1]
```diff
@@ -31,16 +31,20 @@
...
 use OCP\IDBConnection;
-use function array_map;
 
 class MailboxMapper extends QBMapper {
 
+	/**
+	 * @var int
+	 * Lock timeout for sync (5 minutes)
+	 */
+	public const LOCK = 300;
```
Comment (ChristophWurst) [inline lib/Db/MailboxMapper.php:46]: I think the Mailbox entity is a better fit

**Comment by GretaD** [comment]
While testing i realized that things are a better but with the combined mailboxes(in my case 2) the problem still exist if the initial sync is in progress.

File: lib/Db/MailboxMapper.php [hunk 2]
```diff
@@ -119,7 +115,6 @@ public function findById(int $id): Mailbox {
 	 *
 	 * @return Mailbox
 	 *
-	 * @throws DoesNotExistException
```
Comment (ChristophWurst) [inline lib/Db/MailboxMapper.php:122]: please revert

File: lib/Db/MailboxMapper.php [hunk 3]
```diff
@@ -93,7 +90,6 @@ public function find(Account $account, string $name): Mailbox {
...
```
Comment (ChristophWurst) [inline lib/Db/MailboxMapper.php:96]: revert

File: lib/Db/MailboxMapper.php [hunk 4]
```diff
@@ -66,7 +64,6 @@ public function findAll(Account $account): array {
 	}
 
 	/**
-	 * @throws DoesNotExistException
```
Comment (ChristophWurst) [inline lib/Db/MailboxMapper.php:69]: revert

File: lib/Db/MailboxMapper.php [hunk 5]
```diff
@@ -31,13 +31,11 @@
 use OCA\Mail\Account;
 use OCA\Mail\Exception\MailboxLockedException;
 use OCA\Mail\Exception\ServiceException;
-use OCP\AppFramework\Db\DoesNotExistException;
```
Comment (ChristophWurst) [inline lib/Db/MailboxMapper.php:34]: revert

File: lib/Db/MailboxMapper.php [hunk 6]
```diff
@@ -31,13 +31,11 @@
...
 use OCP\AppFramework\Db\MultipleObjectsReturnedException;
 use OCP\AppFramework\Db\QBMapper;
 use OCP\AppFramework\Utility\ITimeFactory;
 use OCP\DB\QueryBuilder\IQueryBuilder;
 use OCP\IDBConnection;
-use function array_map;
```
Comment (ChristophWurst) [inline lib/Db/MailboxMapper.php:40]: still used -> revert

**Review by ChristophWurst — APPROVED** [review]
Looks good

**Pull request merged**